sentry_sdk/tracing.py

```diff
@@ -183,29 +221,77 @@ def to_traceparent(self):
         return "%s-%s-%s" % (self.trace_id, self.span_id, sampled)
 
     def to_legacy_traceparent(self):
+        # type: () -> str
         return "00-%s-%s-00" % (self.trace_id, self.span_id)
 
     def set_tag(self, key, value):
+        # type: (str, Any) -> None
         self._tags[key] = value
 
     def set_data(self, key, value):
+        # type: (str, Any) -> None
         self._data[key] = value
 
     def set_failure(self):
+        # type: () -> None
         self.set_tag("status", "failure")
 
     def set_success(self):
+        # type: () -> None
         self.set_tag("status", "success")
 
     def is_success(self):
+        # type: () -> bool
         return self._tags.get("status") in (None, "success")
 
-    def finish(self):
-        self.timestamp = datetime.now()
-        if self._finished_spans is not None:
-            self._finished_spans.append(self)
+    def finish(self, hub=None):
+        # type: (Optional[sentry_sdk.Hub]) -> Optional[str]
+        hub = hub or self.hub or sentry_sdk.Hub.current
+
+        if self.timestamp is None:
```

Comment:
I think it should short-circuit here, so same transaction cannot be sent multiple times.

```
if self.timestamp is not None:
  return
```

Reply:
I agree, let's do that